services/src/test/scala/cromwell/services/auth/impl/GithubAuthVendingActorSpec.scala

```diff
@@ -8,94 +8,81 @@ import cromwell.services.auth.GithubAuthVending.{
-      )
-    }
-  }
+  private def githubAuthConfigWithoutEcmUrl(flag: Boolean): Config =
+    ConfigFactory.parseString(s"""
+                                 |enabled = $flag
+                                 |auth.azure = $flag
+        """.stripMargin)
 
-  it should "return Github token if found" in {
-    val serviceRegistryActor = TestProbe()
-    val actor = system.actorOf(
-      Props(
-        new TestGithubAuthVendingActor(githubAuthEnabledServiceConfig,
-                                       ConfigFactory.parseString(""),
-                                       serviceRegistryActor.ref
-        )
-      )
+  private val testCases = Table(
+    ("test name", "service config", "use TestEcmService class", "terra token", "response"),
+    ("return NoGithubAuthResponse if GithubAuthVending is disabled",
+     githubAuthConfigWithoutEcmUrl(false),
+     false,
+     validUserToken,
+     NoGithubAuthResponse
+    ),
+    ("return invalid configuration error if no ECM base url is found",
+     githubAuthConfigWithoutEcmUrl(true),
+     false,
+     validUserToken,
+     GithubAuthVendingFailure("Invalid configuration for service 'GithubAuthVending': missing 'ecm.base-url' value.")
+    ),
+    ("return Github token if found",
+     githubAuthEnabledServiceConfig,
+     true,
+     validUserToken,
+     GithubAuthTokenResponse(GithubToken("gha_token"))
+    ),
+    ("return failure message if ECM service returns non-successful response",
+     githubAuthEnabledServiceConfig,
+     true,
+     "invalid_user_token",
+     GithubAuthVendingFailure("Exception thrown for testing purposes")
     )
+  )
 
-    eventually {
-      serviceRegistryActor.send(actor, GithubAuthRequest("valid_user_token"))
-      serviceRegistryActor.expectMsg(GithubAuthTokenResponse("gha_token"))
-    }
-  }
+  behavior of "GithubAuthVendingActor"
 
-  it should "return failure message if ECM service returns non-successful response" in {
-    val serviceRegistryActor = TestProbe()
-    val actor = system.actorOf(
-      Props(
-        new TestGithubAuthVendingActor(githubAuthEnabledServiceConfig,
-                                       ConfigFactory.parseString(""),
-                                       serviceRegistryActor.ref
-        )
+  forAll(testCases) { (testName, serviceConfig, useTestEcmService, userTerraToken, expectedResponseMsg) =>
+    it should testName in {
+      val serviceRegistryActor = TestProbe()
+      val actor = system.actorOf(
+        Props(new TestGithubAuthVendingActor(serviceConfig, serviceRegistryActor.ref, useTestEcmService))
       )
-    )
 
-    eventually {
-      serviceRegistryActor.send(actor, GithubAuthRequest("invalid_user_token"))
-      serviceRegistryActor.expectMsg(GithubAuthVendingFailure("Exception thrown for testing purposes"))
+      eventually {
+        serviceRegistryActor.send(actor, GithubAuthRequest(TerraToken(userTerraToken)))
+        serviceRegistryActor.expectMsg(expectedResponseMsg)
+      }
```

Comment:
You probably want to move the `send` outside the eventually. Otherwise every time the eventually runs, another message is sent

Reply:
done!